### ISSUE TITLE
训练结果不同gs的异常

### ISSUE THREAD

**Issue by trikim** [Zheng-Chong/CatVTON#17]
图像由左到右分别是服装、cloth guidance scale分别为1.0、1.5、2.0、2.5的生成图。

**Comment by Zheng-Chong** [comment]
训练前期可能会有类似的色差问题，也可能是数据预处理或者超参数设置原因。

**Comment by trikim** [comment]
> 训练前期可能会有类似的色差问题，也可能是数据预处理或者超参数设置原因。

感谢大佬的解答。
我后面发现gs增大时变暗变黑的一个主要原因是，在dropout时，对与带mask模特图拼接的服装latents置零的同时，也把noisy latents的右半部分（对应服装的latents）也置零导致的。修改这个后肤色有所改善。
在复现DREAM采样策略的时候，有两个问题想咨询下大佬。
我的训练过程是这样的，假设模特图+服装图拼接特征为latents，随机标准高斯噪声为noise，采样时间步为t，DREAM融合参数为lambda=10
第一次推理：使用noise和t对latents加噪得到noisy_latents，与其他latents在通道维度拼接，unet预测的噪声model_pred1
融合噪声noise2=noise + lambda * model_pred1
第二次推理：使用noise2和t对latents加噪得到noisy_latents2，与其他latents在通道维度拼接，unet预测的噪声model_pred2
计算loss：noise2与model_pred2的MSE Loss
这个过程中我主要有两个疑问：
1）融合噪声时lambda乘的是model_pred1还是（noise - model_pred1）?因为看原DREAM论文，好像写的是noise - model_pred1;
2)训练时第一次推理过程是否需要加torch.no_grad()?原DREAM论文写了StopGradient，但这种方式训得的模型似乎并不能收敛；

**Comment by Zheng-Chong** [comment]
DREAM 在 Diffusers 库中的[训练示例](https://github.com/huggingface/diffusers/blob/3ece14330889139ffad0ce9ac27f46eb353ef08e/examples/text_to_image/train_text_to_image.py#L991)已经实现，DREAM过程在训练时只是对加躁 latent 和 目标噪声进行修正，需要添加`torch.no_grad()`。

**Comment by ZhangLei999** [comment]
@Zheng-Chong DREAM在SSIM和LPIPS指标上效果有轻微的提升，但是在FID和KID指标上不太行，差的有点多，不如不加DREAM策略训练出来的CATVTON模型。这种情况正常吗

**Comment by Zheng-Chong** [comment]
> @Zheng-Chong DREAM在SSIM和LPIPS指标上效果有轻微的提升，但是在FID和KID指标上不太行，差的有点多，不如不加DREAM策略训练出来的CATVTON模型。这种情况正常吗

正常，不加 DREAM 高频细节会更清晰，但是可能造成一些不自然和伪影。

**Comment by ZhangLei999** [comment]
@Zheng-Chong 感谢回答。我在VITONHD数据集上进行，控制变量测试：如果在原始CATVTON基础上用了cfg策略，kid是0.66，如果在原始CATVTON基础上用了dream策略，kid是2；你的论文上kid能达到0.4。我的λ设置的是1，是跟这个超参数有关吗。两个策略一起我还没进行测试，我感觉即使一起用也达不到这么低

**Comment by Zheng-Chong** [comment]
> @Zheng-Chong 感谢回答。我在VITONHD数据集上进行，控制变量测试：如果在原始CATVTON基础上用了cfg策略，kid是0.66，如果在原始CATVTON基础上用了dream策略，kid是2；你的论文上kid能达到0.4。我的λ设置的是1，是跟这个超参数有关吗。两个策略一起我还没进行测试，我感觉即使一起用也达不到这么低

需要一起使用，λ设置为 1 会生成过度平滑的结果，建议 10～20 之间，可以查看论文表4，λ=1时KID=2.1977

**Comment by ZhangLei999** [comment]
收到，谢谢

**Comment by dhmtdhkl** [comment]
你好能分享一下训练脚本吗
